internal/controller/configmap_controller.go

```diff
@@ -156,21 +158,37 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	}
 
-	err := cmlib.AnnotateCM(ctx, r.Client, configMap, internal.InsightsUpdatedAnnotationLabel, time.Now().UTC().Format(time.RFC3339))
-
-	if err != nil {
-		log.Error(err, "Unable to update configmap")
-		return ctrl.Result{}, err
-	}
-
 	// Here we run the post processors
+	retryPostprocessing := false
 	for _, processor := range r.PostProcessors.PostProcessors {
 		err := processor.PostProcess(insightsMessage)
 		if err != nil {
 			log.Error(err, "Post processor failed")
+			retryPostprocessing = true
 		}
 	}
 
+	if retryPostprocessing {
+		//In this case what we want to do is defer the processing to a couple minutes from now
```

Comment:
So I think that it would be great to take this and DRY it out a little. Not sure if it makes sense to move the labelling into a function (although it might) - but I think just moving the future and futuretime calcs into funcs or consts might be cool.